onnx/backend/test/cpp/driver_test.cc

```diff
@@ -241,14 +248,21 @@ TEST_P(ONNXCppDriverTest, ONNXCppDriverUnitTest){
   onnxifi_library lib;
   onnxBackendID backendID = NULL;
   onnxBackend backend;
+  const size_t max_info_size = 50; // The maximum backend info size
   if (!ONNXIFI_DUMMY_BACKEND) {
     ASSERT_TRUE(onnxifi_load(1, NULL, &lib));
     size_t numBackends = 0;
     lib.onnxGetBackendIDs(&backendID, &numBackends);
     const uint64_t backendProperties[] = {ONNXIFI_BACKEND_PROPERTY_NONE};
     lib.onnxInitBackend(backendID, backendProperties, &backend);
+    char infovalue[max_info_size];
```

Comment:
how does this work?

```
onnxBackendID backendID = NULL;
...
size_t numBackends = 0;
    lib.onnxGetBackendIDs(&backendID, &numBackends);
```
In case backendID == nullptr, ONNXIFI implementation will just populate numBackends to be equal to number of supported backends.

Don't you need to call `onnxGetBackendIDs` again to actually initialize backendID?

Reply:
What @rdzhabarov said. Normally, we need two passes. First to get the number of backends. And second to query.